File: Java/CodingBat/cigarParty.java
// http://codingbat.com/prob/p159531

public boolean cigarParty(int cigars, boolean isWeekend) {
  boolean partyIsSuccessful;
  
  boolean cigarsInRange = (cigars >= 40 && cigars <= 60);
  if (cigarsInRange || (isWeekend && cigars >= 40)) {
    partyIsSuccessful = true;
  }
  else {
    partyIsSuccessful = false;
  }

  return partyIsSuccessful;
}

File: Python/CodingBat/cigar_party.py
# http://codingbat.com/prob/p195669

def cigar_party(cigars, is_weekend):
  if is_weekend and cigars >= 40:
    return True
  elif not is_weekend and (cigars >= 40 and cigars <= 60):
    return True
  else:
    return False
